src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,77 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String DATA_SEPARATOR = ",";
+    private static final int OPERATION_TYPE_COLUMN_INDEX = 0;
+    private static final int AMOUNT_COLUMN_INDEX = 1;
+    private static final String SUPPLY_STRING = "supply";
+    private static final String BUE_STRING = "buy";
+    private static final String RESULT_STRING = "result";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final String EMPTY_STRING = "";
+    private StringBuilder stringBuilder = new StringBuilder();
+    private int supplyAmount = 0;
+    private int buyAmount = 0;
 
-    public void getStatistic(String fromFileName, String toFileName) {
+    private void readDataFromFile(String fromFileName) {
+        File file = new File(fromFileName);
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            String readedLine = bufferedReader.readLine();
+            String lineSeparator = EMPTY_STRING;
+            while (readedLine != null) {
+                stringBuilder.append(lineSeparator).append(readedLine);
+                readedLine = bufferedReader.readLine();
+                lineSeparator = LINE_SEPARATOR;
+            }
```

Comment:
can we parse the string in this loop to avoid additional iterating over the strings in the method ```calculateSums()```?

Reply:
ok, I looked again at common errors and there is a point about [Divide program logic](https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#divide-program-logic). so if you relied on this logic, do not respond to this comment